File: lectures/lecture_02/hello_mpi.f90
program hello_mpi
    !use mpi

    implicit none
    include "mpif.h"

    integer :: rank, size, ierr

    ! Initialize MPI
    call mpi_init(ierr)
    call mpi_comm_size(mpi_comm_world,size,ierr)
    call mpi_comm_rank(mpi_comm_world,rank,ierr)

    write(6,*) "Hello world! I am process ", rank, " of " , size

    ! Finalize MPI
    call mpi_finalize(ierr)

end program hello_mpi

File: lecture_02/hello_mpi.cpp
# include<mpi.h>
# include<iostream>

int main(int argc, char** argv){
    int rank, size;

    // Initialize MPI
    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD, &size);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);

    std::cout << "Hello, World! I am " << rank << " of " << size << std::endl;

    // Finalize MPI
    MPI_Finalize();

    return 0;
}
